roles/ocm-install-core/tasks/main.yml

```diff
@@ -34,26 +73,68 @@
       reason: 'AllCatalogSourcesHealthy'
       status: 'False'
 
-- name: Wait for MultiClusterHub (MCH) CRD
-  kubernetes.core.k8s_info:
-    kubeconfig: "{{ ocm_hub_kubeconfig }}"
-    kind: CustomResourceDefinition
-    name: multiclusterhubs.operator.open-cluster-management.io
-    wait: true
-    wait_condition:
-      type: 'Established'
-      reason: 'InitialNamesAccepted'
-      status: 'True'
-    wait_timeout: 300
+- name: Confirm Install of MultiClusterHub
+  block:
+    - name: "[Install] Wait for MultiClusterHub (MCH) CRD"
+      kubernetes.core.k8s_info:
+        kubeconfig: "{{ ocm_hub_kubeconfig }}"
+        kind: CustomResourceDefinition
+        name: multiclusterhubs.operator.open-cluster-management.io
+        wait: true
+        wait_condition:
+          type: 'Established'
+          reason: 'InitialNamesAccepted'
+          status: 'True'
+        wait_timeout: 300
+
+    - name: "[Install] Setup MultiClusterHub (MCH) instance"
+      kubernetes.core.k8s:
+        state: present
+        kubeconfig: "{{ ocm_hub_kubeconfig }}"
+        template: multiclusterhub.yml
+        wait: true
+        wait_condition:
+          type: 'Complete'
+          reason: 'ComponentsAvailable'
+          status: 'True'
+        wait_timeout: 600
+  when: not upgrading
+
+- name: Confirm Upgrade of MultiClusterHub
+  block:
+    - name: "[Upgrade] Wait until old OCM Operator Removed"
+      kubernetes.core.k8s_info:
+        kubeconfig: "{{ ocm_hub_kubeconfig }}"
+        api_version: operators.coreos.com/v1alpha1
+        kind: ClusterServiceVersion
+        name: "{{ ocm_installedCSV }}"
+        namespace: open-cluster-management
+      register: ocm_old_operator
+      until: ocm_old_operator.resources | length == 0
+      retries: 15
+      delay: 20
+
+    - name: "[Upgrade] Wait for MultiClusterHub (MCH) instance"
+      kubernetes.core.k8s_info:
+        kubeconfig: "{{ ocm_hub_kubeconfig }}"
+        api_version: operator.open-cluster-management.io/v1
+        kind: MultiClusterHub
+        name: multiclusterhub
+        namespace: open-cluster-management
+        wait: true
+        wait_condition:
+          type: 'Complete'
+          reason: 'ComponentsAvailable'
+          status: 'True'
+        wait_timeout: 600
+  when: upgrading
 
-- name: Setup MultiClusterHub (MCH) instance
+- name: "Set Plan Approval"
+  set_fact:
+    ocm_plan_approval: "{{ ocm_lockversion | ternary('Manual', 'Automatic', 'Manual') }}"
```

Comment:
we have similar section in `- name: Plan Approval Automatic for Install/Upgrade`

Reply:
This is where the code respects the user's provided value (after operator is setup).